lib/provider.ex

```diff
@@ -182,6 +182,28 @@ defmodule Provider do
         |> Keyword.fetch!(:params)
         |> Enum.map(fn {name, spec} -> {name, quote(do: %{unquote_splicing(spec)})} end)
 
+      use GenServer
+
+      def start_link do
+        GenServer.start_link(__MODULE__, :ok, name: __MODULE__)
+      end
+
+      @impl GenServer
+      def init(:ok) do
+        Provider.Cache.new(__MODULE__)
+
+        load!()
+
+        {:ok, nil}
+      end
+
+      def child_spec(_arg) do
+        %{
+          id: __MODULE__,
+          start: {__MODULE__, :start_link, []}
+        }
```

Comment:
Nit: `use GenServer` already generates `child_spec`, the only diff is that it invokes `start_link/1`. So if you had something like:

```elixir
def start_link(arg), do: GenServer.start_link(__MODULE__, arg, ...)

def init(_arg), do: ...
```

Then you wouldn't need to define `child_spec` yourself.